### PR TITLE
Adding Test Cases for all Models as well as Profiles and Post Controllers

### PR DESCRIPTION
**Pull request by shujaamarwat24**
* comment_spec.rb: Added and improved the tests for comment model
* followship_spec.rb: Added and improved the tests for followship model
* like_spec.rb: Added and improved the tests for Like model
* story_spec.rb: Added and improved the tests for story model
* user_spec.rb: Added tests for followship and user association 
* profiles_controller.rb: Made changes to reject_request method
* post_policy: Added changes to post policy
* story_policy: Added changes to story policy
* user_policy.rb: Added changes to reject_request method policy
* rails_helper.rb: Added config option for requests in devise
* posts_spec: Added tests for post controller
* profiles_spec: Added tests for profiles controller

### CODE REVIEW

File: app/controllers/profiles_controller.rb [hunk 1]
```diff
@@ -23,7 +23,7 @@ def follow
     elsif current_user.followees.push(@user)
       if !@user.private
         flash[:notice] = 'Followed'
-        Followship.is_pending(current_user, @user).update(status: 'Accepted')
+        @pending.update(status: 'Accepted')
```
Comment (MuhammadMusfir) [inline app/controllers/profiles_controller.rb:26]: Missing failure handling for @pending.update

File: app/controllers/profiles_controller.rb [hunk 2]
```diff
@@ -36,7 +36,7 @@ def follow
 
   def accept_request
     if !request_accepted?(@user) || !request_accepted?(@followee)
-      Followship.is_pending(current_user, @user).update(status: 'Accepted')
+      @pending.update(status: 'Accepted')
```
Comment (MuhammadMusfir) [inline app/controllers/profiles_controller.rb:39]: Missing failure handling for @pending.update

File: spec/controllers/posts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+RSpec.describe 'Posts', type: :request do
+  describe 'GET /new' do
+    context 'when user is NOT signed in' do
+      let!(:test_user) { create(:user) }
+
+      it 'returns a failure message' do
+        get new_post_path
+        expect(response.status).to eq(302)
+        expect(flash[:alert]).to eq('You need to sign in or sign up before continuing.')
```
Comment (MuhammadMusfir) [inline spec/controllers/posts_spec.rb:11]: We don't check/match flash messages usually.

**Review by MuhammadMusfir — CHANGES_REQUESTED** [review]

File: spec/models/post_spec.rb [hunk 4]
```diff
@@ -1,25 +1,27 @@
...
 
   describe 'validations' do
     it { is_expected.to validate_presence_of(:user_id) }
-    it { is_expected.to validate_length_of(:description).is_at_least(1).is_at_most(150)}
+    it { is_expected.to validate_length_of(:description).is_at_least(1).is_at_most(150) }
   end
 
   describe 'custom validations' do
-    let(:user1) {
+    let(:user1) do
```
Comment (MuhammadMusfir) [inline spec/models/post_spec.rb:19]: user1 is a bad variable name.

**Review by MuhammadMusfir — APPROVED** [review]

File: app/controllers/profiles_controller.rb (hunk 1 above)
Comment (shujaamarwat24) [reply app/controllers/profiles_controller.rb]: Incorporated

File: app/controllers/profiles_controller.rb (hunk 2 above)
Comment (shujaamarwat24) [reply app/controllers/profiles_controller.rb]: Incorporated

File: spec/controllers/posts_spec.rb (hunk 3 above)
Comment (shujaamarwat24) [reply spec/controllers/posts_spec.rb]: The problem is that simplecov checks the flash messages and checks to add these to the covered lines

**Pull request merged**